### ISSUE TITLE
SAML2PSecurityTokenHandler configuration

### ISSUE THREAD

**Issue by Raschmann** [Sustainsys/Saml2#145]
Add support for `<securityTokenHandlerConfiguration>` configurations:
- audienceUris
- certificateValidation
- certificateValidationMode
- issuerNameRegistry

http://msdn.microsoft.com/en-us/library/hh568639(v=vs.110).aspx

Add support for certificate validation base on the provided configuration

**Comment by explunit** [comment]
Am I correct that this enhancement (if implemented), would let me validate by `trustedIssuers` `thumbprint`, e.g. as shown here: https://stackoverflow.com/questions/28845076/wif-config-issuernameregistry-vs-certificatevalidation

**Comment by AndersAbel** [comment]
@explunit Fixing this would allow a custom `IssuerNameRegistry` to be used, but to go all the way and only have the thumbprint in the configuration would probably require more work. Then the actual certificate used to validate the signature would have to be extracted from the signature and checked for validity through the thumbprint.

Based on the feedback in this issue and from other places, I'm planning to rework how AuthServices integrates with WIF to allow for more of WIF's extensibility and configuration points to use, but it's quite a big job and requires rewriting of a lot of the core code.

**Comment by AndersAbel** [comment]
I've been working on the rewrite mentioned above, and unfortunately I have some bad news: It won't work.

What I tried to do was to replace the entire certificate validation with WIF's method. Unfortunately I can't get it working and as far as I can tell the WIF signature validation doesn't support XML containing white spaces. This means that for now I'm dropping that idea and instead heading for extending our own signature handling. I've commited what I did to the [UseWIFSignatureValidation branch](https://github.com/AndersAbel/authservices/tree/UseWIFSignatureValidation) if anyone else wants to have a look.
